homeassistant/components/websocket_api/commands.py

```diff
@@ -213,30 +220,34 @@ def handle_unsubscribe_events(
         vol.Required("service"): str,
         vol.Optional("target"): cv.ENTITY_SERVICE_FIELDS,
         vol.Optional("service_data"): dict,
+        vol.Optional("return_response", default=False): bool,
     }
 )
 @decorators.async_response
 async def handle_call_service(
     hass: HomeAssistant, connection: ActiveConnection, msg: dict[str, Any]
 ) -> None:
     """Handle call service command."""
-    blocking = True
     # We do not support templates.
     target = msg.get("target")
     if template.is_complex(target):
         raise vol.Invalid("Templates are not supported here")
 
     try:
         context = connection.context(msg)
-        await hass.services.async_call(
+        response = await hass.services.async_call(
             msg["domain"],
             msg["service"],
             msg.get("service_data"),
-            blocking,
+            True,
             context,
             target=target,
+            return_response=msg["return_response"],
         )
-        connection.send_result(msg["id"], {"context": context})
+        result: dict[str, Context | ServiceResponse] = {"context": context}
+        if msg["return_response"]:
+            result["response"] = response
```

Comment:
Should we name the response item just `"response"` as it is now or rename to `"service_response"` or even `"service_response_data"`?

Reply:
We already use `response` in the `execute_script` function. We should be consistent, and if we want to change this to be `service_response` then we should change it there as well, at least IMO

https://github.com/raman325/home-assistant/blob/08f074f634c1a47b2a1f34aed355c186518d7ed3/homeassistant/components/websocket_api/commands.py#L782-L788

Reply:
Ok, let's keep it then.